src/AbstractSQLOptimiser.ts

```diff
@@ -650,6 +650,7 @@ const typeRules: Dictionary<MatchFn> = {
 	Floor: matchArgs('Floor', NumericValue),
 	Ceiling: matchArgs('Ceiling', NumericValue),
 	ToDate: matchArgs('ToDate', DateValue),
+	DateTrunc: matchArgs('DateTrunc', Field, _.identity),
```

Comment:
I think this can operate on any date value and not just fields right?
```suggestion
	DateTrunc: matchArgs('DateTrunc', DateValue, _.identity),
```
And the precision value needs to be text right, so it can have improved validation via
```suggestion
	DateTrunc: matchArgs('DateTrunc', DateValue, TextValue),
```
Also I think it'd be best to keep the order the same as the postgres function since it's basically a 1-1 mapping and having it switched could make it more confusing to use (although I agree in a vacuum I'd also go `date, precision` rather than `precision, date`)
```suggestion
	DateTrunc: matchArgs('DateTrunc', TextValue, DateValue),
```